python/pyarrow/src/arrow/python/inference.cc

```diff
@@ -560,6 +573,17 @@ class TypeInferrer {
     return Status::OK();
   }
 
+  Status VisitArrowScalar(PyObject* obj, bool* keep_going) {
```

Comment:
I think we can add an `/* unused */` comment for `keep_going` since we do want to check all values.